client/model.go

```diff
@@ -40,6 +40,73 @@ type ProjectCreatePayload struct {
 	Description string `json:"description"`
 }
 
+type EnvironmentCreate struct {
+	Name                        string         `json:"name"`
+	ProjectId                   string         `json:"projectId"`
+	DeployRequest               *DeployRequest `json:"deployRequest"`
+	WorkspaceName               string         `json:"workspaceName,omitempty"`
+	RequiresApproval            bool           `json:"requiresApproval,omitempty"`
+	ContinuousDeployment        bool           `json:"continuousDeployment,omitempty"`
+	PullRequestPlanDeployments  bool           `json:"pullRequestPlanDeployments,omitempty"`
+	AutoDeployOnPathChangesOnly bool           `json:"autoDeployOnPathChangesOnly,omitempty"`
+	AutoDeployByCustomGlob      bool           `json:"autoDeployByCustomGlob,omitempty"`
+}
+
+type DeployRequest struct {
+	BlueprintId          string                `json:"blueprintId,omitempty"`
+	BlueprintRevision    string                `json:"blueprintRevision,omitempty"`
+	BlueprintRepository  string                `json:"blueprintRepository,omitempty"`
+	ConfigurationChanges *ConfigurationChanges `json:"configurationChanges,omitempty"`
+	TTL                  *TTL                  `json:"ttl,omitempty"`
+	EnvName              string                `json:"envName,omitempty"`
+	UserRequiresApproval bool                  `json:"userRequiresApproval,omitempty"`
+	Targets              string                `json:"targets,omitempty"`
+	PrPlanOnly           bool                  `json:"prPlanOnly,omitempty"`
+	// TODO: not sure about this one
+	CustomEnv0EnvironmentVariables *interface{} `json:"customEnv0EnvironmentVariables,omitempty"`
+	GitUserData                    *GitUserData `json:"gitUserData,omitempty"`
```

Comment:
Same as pr plan 
only for VCS events we can remove that

Reply:
63d1a3c